eda/src/autogluon/eda/analysis/shift.py

```diff
@@ -0,0 +1,347 @@
+    split : float, default = 0.5
+        Training/test split proportion for classifier 2 sample test
+    classifier_kwargs : dict, default = {}
+        The kwargs passed to the classifier, a member of classifier_class
+    """
+    def __init__(self,
+                 classifier_class,
+                 sample_label='xshift_label',
+                 eval_metric=balanced_accuracy,
+                 split=0.5,
+                 compute_fi = True,
+                 classifier_kwargs = {}
+                 ):
+        classifier_kwargs.update({'label': sample_label, 'eval_metric': eval_metric})
+        self.classifier = classifier_class(**classifier_kwargs)
+        self.classifier_class = classifier_class
+        self.split = split
+        self.sample_label = sample_label
+        self.eval_metric = eval_metric
+        self._is_fit = False
+        self._test = None
+        self.test_stat = None
+        self.has_fi = None
+        self.compute_fi = compute_fi
+
+    @staticmethod
+    def _make_source_target_label(data, sample_label):
+        """Turn a source, target pair into a single dataframe with label column"""
+        source, target = data[0].copy(), data[1].copy()
+        source.loc[:,sample_label] = 0
+        target.loc[:,sample_label] = 1
+        data = pd.concat((source, target))
+        return data
+
+    def fit(self, data, **kwargs):
+        """Fit the classifier for predicting if source or target and compute the 2-sample test statistic.
+
+        Parameters
+        ----------
+        data : pd.DataFrame, or tuple
+            either
+            - a dataframe with a label column where 1 = target and 0 = source
+            - a tuple of source dataframe and target dataframe
+        """
+        if isinstance(data, pd.DataFrame):
+            sample_label = self.sample_label
+            assert sample_label in data.columns, "sample_label needs to be a column of data"
+            assert self.split, "sample_label requires the split parameter"
+            data = data.copy() # makes a copy
+        else:
+            assert len(data) == 2, "Data needs to be tuple/list of (source, target) if sample_label is None"
+            data = self._make_source_target_label(data, self.sample_label) # makes a copy
+        if data.index.has_duplicates:
+            data.index = pd.RangeIndex(data.shape[0])
+        train = data.sample(frac=self.split)
+        test = data.drop(train.index)
+        self.classifier.fit(train, **kwargs)
+        yhat = self.classifier.predict(test)
+        self.test_stat = self.eval_metric(test[self.sample_label], yhat)
+        self.has_fi = (getattr(self.classifier, "feature_importance", None) is not None)
+        if self.has_fi and self.compute_fi:
+            self._test = test  # for feature importance
+        self._is_fit = True
+
+    @post_fit
+    def _pvalue_half_permutation(self,
+                                 num_permutations=1000):
```

Comment:
do we need `1000` rounds here? What is the number of measurements we need to get the statistically-significant results?